flower/command.py

```diff
@@ -103,6 +110,33 @@ def warn_about_celery_args_used_in_flower_command(ctx, flower_args):
         )
 
 
+def is_broker_connected(celery_app):
+    is_connected = False
+    max_retries = celery_app.conf.broker_connection_max_retries
+
+    if not celery_app.conf.broker_connection_retry:
+        max_retries = 0
+
+    with celery_app.connection() as conn:
+        broker_url = conn.as_uri()
+
+        def _error_handler(exc, interval):
+            next_step = f"Trying again {humanize_seconds(interval, 'in', ' ')}... ({int(interval / 2)}/{max_retries})"
+            logger.error(f'Cannot connect to broker: {broker_url}. Error: {exc}. {next_step}')
+
+        try:
+            conn.ensure_connection(errback=_error_handler, max_retries=max_retries)
```

Comment:
Taken from the way celery does it so all the credit goes to them :): https://github.com/celery/celery/blob/master/celery/worker/consumer/consumer.py#L435

Kombu code for reference: https://docs.celeryproject.org/projects/kombu/en/stable/_modules/kombu/connection.html#Connection.ensure_connection